### PR TITLE
add useMountedRef hook

### PR DESCRIPTION
**Pull request by AndreyChernykh**
### 🛠 Purpose

- Add `usePrevious` hook.
---

### ✏️ Notes to Reviewer

- Add `@testing-library/react-hooks` 
- Add `packages/helpers/CHANGELOG.md`
---

### 🖥 Screenshots

---

### CODE REVIEW

File: packages/helpers/test/useMountedRef.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import useMountedRef from "../src/hooks/useMountedRef";
+
+describe("useMountedRef", () => {
+  test("should increment counter", () => {
+    const {
+      result: { current: mountedRef },
+      rerender,
+      unmount,
+    } = renderHook(() => {
```
Comment (nahumzs) [inline packages/helpers/test/useMountedRef.spec.js]: ohhh, so this is like you were coding your component and using the hook inside of it?

**Comment by nahumzs** [comment]
I'm not seeing the `CHANGELOG.md` file, did you omit on purpose?

File: packages/helpers/test/useMountedRef.spec.js (hunk 1 above)
Comment (AndreyChernykh) [reply packages/helpers/test/useMountedRef.spec.js]: Sort of

File: packages/helpers/test/useMountedRef.spec.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { renderHook } from "@testing-library/react-hooks";
+
+import useMountedRef from "../src/hooks/useMountedRef";
+
+describe("useMountedRef", () => {
+  test("should increment counter", () => {
```
Comment (mikrotron) [inline packages/helpers/test/useMountedRef.spec.js]: I'm confused by the description "should increment counter". Was it copied from somewhere else?

Comment (AndreyChernykh) [reply packages/helpers/test/useMountedRef.spec.js]: Oh no! I'll fix it

File: packages/helpers/src/hooks/useMountedRef.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  const mountedRef = React.useRef(false);
+
+  React.useEffect(() => {
+    mountedRef.current = true;
+    return function cleanup() {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  return mountedRef;
```
Comment (alexzherdev) [inline packages/helpers/src/hooks/useMountedRef.js]: Late to the game, but why not return `mountedRef.current` here? The end user probably never cares about the ref itself, they're interested in the boolean value of whether we're mounted or not (and so `ref` in the hook name looks irrelevant to me)

Comment (AndreyChernykh) [reply packages/helpers/src/hooks/useMountedRef.js]: @mikrotron @alexzherdev 

Ref is safer to use with effects. Let's say you want to make a request `on first mount`: 

```js
useEffect(() => {
  fetch().then(() => {
    if (ref.current) { // do something }
  })
}, [ref]);
```

Since `ref` is immutable, the change will not trigger an effect to reapply. 

But if we do `const isMounted = useMounted()`

then we have to pass it as parameter. And the effect will be reapplied when the component dismounts.  

```js
useEffect(() => {
  fetch().then(() => {
    if (isMounted) { // do something }
  })
}, [isMounted]);
```

Comment (alexzherdev) [reply packages/helpers/src/hooks/useMountedRef.js]: Hmm I see. In the former case, does eslint warn if you don't specify `ref` as dependency? I think refs are guaranteed to keep referential integrity.
I'm assuming in the latter case, if you don't specify `isMounted`, it should definitely warn.

Comment (AndreyChernykh) [reply packages/helpers/src/hooks/useMountedRef.js]: @alexzherdev Yes, it'll warn you. 
But with `ref` it is safe to add it as a dependency because of `referential integrity`. 

We might think of having a warning instead of an error (as we have now) for that. But I'm worried that people might ignore warnings 🤔
